test/ng/compileSpec.js

```diff
@@ -3742,6 +3742,34 @@ describe('$compile', function() {
       });
 
 
+      it('should trigger `$onChanges` even if the inner value already equals the new outer value', function() {
+        var log = [];
+        function TestController() { }
+        TestController.prototype.$onChanges = function(change) { log.push(change); };
+
+        angular.module('my', [])
+          .component('c1', {
+            controller: TestController,
+            bindings: { 'prop1': '<' }
+          });
+
+        module('my');
+        inject(function($compile, $rootScope) {
+          $rootScope.getVal = function() { return $rootScope.val; };
+          element = $compile('<c1 prop1="val"></c1>')($rootScope);
+          var isolateScope = element.isolateScope();
+
+          log = [];
+          $rootScope.$apply('val = 1');
+          expect(log.pop()).toEqual({prop1: jasmine.objectContaining({previousValue: undefined, currentValue: 1})});
+
+          isolateScope.$ctrl.prop1 = 2;
```

Comment:
Nitpick: `element.controller('c1').prop1 = 2` saves one line of code (Yay :disappointed_relieved:)